pkg/controller/storagecluster/storagecluster.go

```diff
@@ -649,6 +658,45 @@ func (c *Controller) deleteStorageCluster(
 	return nil
 }
 
+func (c *Controller) removeResources(namespace string) error {
+	objs, err := k8s.GetAllObjects(c.client, namespace)
+	if err != nil {
+		return err
+	}
+
+	for _, obj := range objs {
+		gcAnnotated := false
+		if obj.GetAnnotations() != nil {
+			gcAnnotated, _ = strconv.ParseBool(obj.GetAnnotations()[constants.AnnotationGarbageCollection])
+		}
+
+		if gcAnnotated || c.gcNeeded(obj) {
+			logrus.Infof("Garbage collect object %s %s", obj.GetObjectKind().GroupVersionKind().Kind, obj.GetName())
+			if err := c.client.Delete(context.TODO(), obj); err != nil {
```

Comment:
need to ignore `NotFound` errors

Reply:
It should not happen as the object is found before deleting it.